x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx

```diff
@@ -111,7 +112,7 @@ export function FieldItem(props: FieldItemProps) {
     core.http
       .post(`/api/lens/index_stats/${indexPattern.title}/field`, {
         body: JSON.stringify({
-          query: toElasticsearchQuery(query, indexPattern),
+          dslQuery: buildEsQuery(indexPattern, query, filters),
```

Comment:
buildEsQuery also takes in an optional config that allows us to use the global Kibana query language (either KQL or lucene) and the timezone users choose to set in their Kibana advanced settings. We have access to those from KibanaContextProvider services.  There are defaults though, so it's optional.